### ISSUE TITLE
Mongo 6 is unsupported

### ISSUE THREAD

**Issue by laurenskling** [keystonejs/keystone-classic#5012]
Atlas is going to upgrade to Mongo 6 pretty soon, and I'm getting errors starting Keystone 4 with Mongo 6.
Just wanted to leave a issue here, see if anyone else is still rocking Keystone 4 and is using Atlas.

**Comment by maheeptathgur-rb** [comment]
any updates here.. my system has ceased working because of this.

**Comment by laurenskling** [comment]
I don't think any work will ever be done again on keystone 4, so I'm afraid you will have to stick to Mongo 5 (or 4)

**Comment by dijs** [comment]
Just ran into this today too... guess this weekend is admin update weekend...

**Comment by quangtruongdit** [comment]
I have got the same issue after Mongo is updated to version 6

`2023-04-17T00:01:36.918079+00:00 app[web.1]: Unsupported OP_QUERY command: listIndexes. The client driver may require an upgrade. For more details see https://dochub.mongodb.org/core/legacy-opcode-removal MongoError: Unsupported OP_QUERY command: listIndexes. The client driver may require an upgrade. For more details see https://dochub.mongodb.org/core/legacy-opcode-removal`

Looking forwards to see any update from keystone for supporting it.

**Comment by laurenskling** [comment]
Development on this project has stopped almost 3 years ago. The chance someone is going to make support for Mongo 6 is very very low. I would advise you to not want to upgrade to Mongo 6.

**Comment by richardvanderdys** [comment]
I am in progress of switching to Keystone 6 and moving data from Mongo to Postgres. It will be a long, slow, process... Unless someone has any better ideas.

**Comment by sefigrossman** [comment]
I had to create a dedicated instance to downgrade to Mongo 5 for now to get my application running until I can refactor my entire application.

**Comment by richardvanderdys** [comment]
@sefigrossman Appreciate the feedback. It has been a real pain to refactor this... can you share a few more details on where and how you got a Mongo 5 instance up?

**Comment by sefigrossman** [comment]
> @sefigrossman Appreciate the feedback. It has been a real pain to refactor this... can you share a few more details on where and how you got a Mongo 5 instance up?


I'm assuming you are on mongodb.com: Take a backup of your last v5 database snapshot. Then create a new dedicated cluster (lowest you can do is M10 at $0.08 an hour I believe). From there it will allow you to downgrade (that option is obviously not possible in a shared cluster). Restore the backup and change your existing connection string to reflect the new cluster name.

<img width="928" alt="Screen Shot 2023-04-20 at 4 58 26 PM" src="https://user-images.githubusercontent.com/297111/233486377-a13ac537-db24-46b5-8601-9a6c5d1d8945.png">

This will buy you time to consider what to do to rebuild your application long term. Costs a little more but better than being down while you ponder.

**Comment by richardvanderdys** [comment]
You are so helpful! Saved my butt. Thank you. @sefigrossman

**Comment by sefigrossman** [comment]
> You are so helpful! Saved my butt. Thank you. @sefigrossman

I'm looking hard at Directus.io as a replacement for Keystone v4. I may look at Keystone v6 as I'm sure they are similar in feature set but I'm very impressed with Directus so far.

**Comment by richardvanderdys** [comment]
Yep. I have used Directus in the past. It is great, but it just got pretty expensive for an entry tier ($99/mo)

**Comment by sefigrossman** [comment]
> 

Negative, it's open source and you can self host. They hide that at the bottom ;)
